sda-download/api/api.md

```diff
@@ -51,8 +51,8 @@ The files contained in a dataset are listed using the `datasetName` obtained fro
 GET /metadata/datasets/{datasetName}/files
 ```
 **Scheme Parameter**
-The `?scheme=` query parameter is optional. When a dataset contains a scheme, it may sometimes encounter issues with reverse proxies.
-The scheme can be separated from the dataset name and supplied in a query parameter.
+The `?scheme=` query parameter is optional. When a dataset name contains a scheme, such as `https://`, it may sometimes encounter issues with reverse proxies.
+This can be solved by separating the scheme from the dataset name and suppling it as a query parameter.
 ```
```

Comment:
It's a bit confusing with the inclusion of the source code here. I think it is better either remove it from the README, i.e. only including the instruction for how to use the `?scheme` API.
If we really want to include how it works, maybe add it as annotating text by 
> Below is a demonstration of how the dataset name with a schema is parsed in the code.
> ```dataset := ...```

Reply:
I agree. I'm tempted to replace it with an example, as below. What do you think?

> This can be solved by separating the scheme from the dataset name and suppling it as a query parameter.
> ```
> GET /metadata/datasets/{datasetName}/files?scheme=https
> ```
> For example, given a dataset name `https://doi.org/abc/123`, one can do `GET /metadata/datasets/doi.org/abc/123/files?scheme=https`.
>

Reply:
You solution looks great!